sympy/matrices/immutable.py

```diff
@@ -158,6 +158,8 @@ def __setitem__(self, *args):
 
     subs = MatrixBase.subs
 
+    xreplace = MatrixBase.xreplace
+
```

Comment:
@certik On the contrary I only added it because the code was failing the tests without! The relevant test is at line 951 in test_matrices.py below, because `classes` includes `ImmutableSparseMatrix`. Note that this test is just a copy of the tests for the preexisting `subs` method.